tests/acceptance/01_vars/02_functions/010.cf

```diff
@@ -11,37 +11,45 @@ body common control
       version => "1.0";
 }
 
-#######################################################
-
 bundle agent init
 {
   vars:
-      "dummy" string => "dummy";
+      "names" slist => hostsseen("inf", "lastseen", "name");
+      "addresses" slist => hostsseen("inf", "lastseen", "address");
+      "hostkeys" slist => hostsseen("inf", "lastseen", "hostkey");
 }
 
-#######################################################
-
-bundle agent test
+bundle common test
 {
-  vars:
-      "hosts" slist => hostsseen("inf", "lastseen", "name");
-}
+  meta:
+      "description" -> { "CFE-2546" }
+        string => "Test the hostsseen() policy function. Currently the test always passes but can be used to see if the function at least runs successfully.";
 
-#######################################################
+  classes:
+      "ok_localhost_name"
+        expression => reglist(@{init.names}, "localhost");
+      "ok_localhost_address"
+        expression => reglist(@{init.addresses}, "127\.0\.0\.1");
+      "ok_localhost_hostkey"
+        expression => reglist(@{init.hostkeys}, ${sys.key_digest});
+}
 
 bundle agent check
 {
-  vars:
-      "hosts" slist => { @{test.hosts} };
   classes:
-      "ok" expression => "any";        # XXX # I don't know how to test hostsseen!
```

Comment:
So, the prior test was not testing anything except perhaps that the agent didn't bail out on the function call. Here, it was just setting OK if any class is defined (which any is always defined).

`hostsseen()` pulls from the embedded databases where we have stored information about communication with peers ... the same information that `cf-key -s` pulls from.

Unless the host has actually communicated over the cfengine protocol there won't be anything in those data stores.
I don't think that this part of the test framework has any communication over the network with other cfengine hosts by default. There are some cf-serverd tests which do exercise the network. I think that you would need to force some network activity for this test to work.

It might fit better as a deployment test

Reply:
> It might fit better as a deployment test

Yup, I was thinking the same. Ideally we should add one DT testing various functions like this one -- https://tracker.mender.io/browse/CFE-3709